core/src/main/java/org/elasticsearch/rest/AbstractRestChannel.java

```diff
@@ -133,4 +148,27 @@ public RestRequest request() {
     public boolean detailedErrorsEnabled() {
         return detailedErrorsEnabled;
     }
+
+    /**
+     * A wrapper around a {@link BytesStreamOutput} that makes the close operation a no-op. This is
+     * needed as closing a {@link XContentBuilder} will close the output stream, which in some
+     * cases releases resources that we still need to use to send a response.
+     */
+    private static class UnclosableBytesStream extends FilterOutputStream implements BytesStream {
+
+        private final BytesStreamOutput delegate;
+
+        private UnclosableBytesStream(BytesStreamOutput bytesStreamOutput) {
+            super(bytesStreamOutput);
+            this.delegate = bytesStreamOutput;
+        }
+
+        @Override
+        public void close() {}
+
+        @Override
+        public BytesReference bytes() {
```

Comment:
maybe I am blind but this seems unused?

I wonder if we should make this class a generic class and remove this method, and call flush on close that way we can use it in several places?